algosdk/template.py

```diff
@@ -301,34 +300,55 @@ def get_program(self):
         types = [int, int, int, int, int, int, "address"]
         return inject(orig, offsets, values, types)
 
-    def get_swap_assets_transactions(self, amount: int, contract: bytes,
+    @staticmethod
+    def get_swap_assets_transactions(contract: bytes, asset_amount: int,
+                                     microalgo_amount: int,
                                      private_key: str, first_valid,
                                      last_valid, gh, fee):
         """
         Return a group transactions array which transfer funds according to
         the contract's ratio.
 
         Args:
-            amount (int): the amount of assets to be sent
-            contract (bytes): the contract containg information, should be
-                recieved from payer
+            contract (bytes): the contract containing information, should be
+                received from payer
+            asset_amount (int): the amount of assets to be sent
+            microalgo_amount (int): the amount of microalgos to be received
             private_key (str): the secret key to sign the contract
             first_valid (int): first valid round for the transactions
             last_valid (int): last valid round for the transactions
             gh (str): genesis hash in base64
             fee (int): fee per byte
         """
-        txn_1 = transaction.PaymentTxn(self.get_address(), fee,
-                                       first_valid, last_valid, gh,
-                                       account.address_from_private_key(
-                                       private_key), int(
-                                           amount/self.ratn*self.ratd))
-
-        txn_2 = transaction.AssetTransferTxn(account.address_from_private_key(
-                                             private_key), fee,
-                                             first_valid, last_valid, gh,
-                                             self.owner, amount,
-                                             self.asset_id)
+        address = logic.address(contract)
+        _, ints, bytearrays = logic.read_program(contract)
+        if not (len(ints) == 10 and len(bytearrays) == 1):
+            raise error.WrongContractError("Wrong contract provided; " +
+                                           "a limit order contract" +
+                                           " is needed")
+        min_trade = ints[4]
+        asset_id = ints[6]
+        ratn = ints[8]
+        ratd = ints[7]
+        owner = encoding.encode_address(bytearrays[0])
+
+        if microalgo_amount < min_trade:
+            raise error.TemplateError("At least " + str(min_trade) +
+                                      " microalgos must be requested")
+
+        if asset_amount*ratd < microalgo_amount*ratn:
```

Comment:
I think it needs to be exactly `asset_amount*ratd == microalgo_amount*ratn`

Reply:
I don't think it needs to be exact, at least based on https://github.com/algorand/go-algorand/blob/master/tools/teal/templates/limit-order-a.teal.tmpl